pkg/scale/encode.go

```diff
@@ -343,10 +470,19 @@ func (es *encodeState) encodeUint(i uint) (err error) {
 	switch {
 	case i < 1<<6:
 		err = binary.Write(es, binary.LittleEndian, byte(i)<<2)
+		if err != nil {
+			err = fmt.Errorf("encoding uint: %w", err)
```

Comment:
The context `encoding uint` should be added by callers of `encodeUint`, not here. However, we should add the `uint` value trying to be encoded if it fails:

```suggestion
			err = fmt.Errorf("writing binary for uint %d: %w", i, err)
```